xbmc/cores/VideoPlayer/DVDCodecs/Video/AMLCodec.cpp

```diff
@@ -2031,20 +2029,23 @@ bool CAMLCodec::OpenDecoder(CDVDStreamInfo &hints)
     }
 
     am_private->gcodec.dv_enable = 1;
-    if (!m_is_dv_p7_mel && (hints.dovi.dv_profile == 4 || hints.dovi.dv_profile == 7) && CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(
+    if ((hints.dovi.dv_profile == 4 || hints.dovi.dv_profile == 7) && CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(
         CSettings::SETTING_VIDEOPLAYER_CONVERTDOVI) == 0)
     {
       CSysfsPath amdolby_vision_debug{"/sys/class/amdolby_vision/debug"};
-      if (amdolby_vision_debug.Exists())
-        amdolby_vision_debug.Set("enable_fel 1");
-      am_private->gcodec.dec_mode  = STREAM_TYPE_STREAM;
 
-      CSysfsPath dolby_vision_wait_delay{"/sys/module/amdolby_vision/parameters/dolby_vision_wait_delay"};
-      if (dolby_vision_wait_delay.Exists())
-      {
-        m_dolby_vision_wait_delay = dolby_vision_wait_delay.Get<unsigned int>().value();
-        CLog::Log(LOGDEBUG, "CAMLCodec::OpenDecoder DoVi P{:d} MEL detection frame delay got set to {:d} frames", hints.dovi.dv_profile, m_dolby_vision_wait_delay);
-      }
+      if (dovi_el_type == 2) // MEL
+      { 
+        am_private->gcodec.dec_mode  = STREAM_TYPE_FRAME;
+        if (amdolby_vision_debug.Exists())
+          amdolby_vision_debug.Set("enable_fel 0");
+      } 
+      else // FEL and not determined
+      { 
+        am_private->gcodec.dec_mode  = STREAM_TYPE_STREAM;
+        if (amdolby_vision_debug.Exists())
+          amdolby_vision_debug.Set("enable_fel 1");
+      }      
```

Comment:
This if/else logic isn't needed. You can add `dovi_el_type` check in the first `if` that checks for the profiles.
If it's MEL, skip that if.

Reply:
It is setting the enable_fel flag differently for FEL / MEL

Can leave out:
am_private->gcodec.dec_mode = STREAM_TYPE_FRAME;
but I left that for clarity for anyone reading the code, to determine the difference in handling in one place.